backend/Backend/Backend/Repository/TimetableRepository.cs

```diff
@@ -0,0 +1,35 @@
+﻿using Backend.Context;
+using Backend.Domain;
+using Backend.Domain.Enums;
+using Backend.Interfaces;
+using Microsoft.EntityFrameworkCore;
+
+namespace Backend.Repository;
+
+public class TimetableRepository(AcademicAppContext context) : ITimetableRepository
+{
+    private readonly AcademicAppContext _context = context;
+    
+    public async Task<Subject> AddSubjectAsync(Subject subject)
+    {
+        await _context.Subjects.AddAsync(subject);
+        return subject;
+    }
+    
+    public async Task<Subject?> GetSubjectByIdAsync(int id)
+    {
+        return await _context.Subjects.SingleOrDefaultAsync(f => f.Id == id);
+    }
+    
+    public async Task<Subject?> GetSubjectByNameAsync(string name)
+    {
+        return await _context.Subjects.SingleOrDefaultAsync(f => f.Name == name);
```

Comment:
```suggestion
        return await _context.Subjects.FirstOrDefaultAsync(f => f.Name == name);
```

Same thing here. We are constraining in the builder to have the name be unique, so just use `First` since it's slightly faster